### PR TITLE
added support for Visual Studio Code (VSC) extension update

### PR DESCRIPTION
**Pull request by AlexanderWillner**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The code compiles
- [x] The code passes rustfmt
- [x] The code passes clippy
- [x] The code passes tests (no tests implemented)
- [x] *Optional:* I have tested the code myself
    - [x] I also tested that Topgrade skips the step where needed

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Can we rename this to VSCode? VSC is not as common and probably confuses a few people what it is.

File: src/steps/generic.rs [hunk 1]
```diff
@@ -85,6 +85,18 @@ pub fn run_sheldon(ctx: &ExecutionContext) -> Result<()> {
...
 )))]
+pub fn run_vscode(run_type: RunType) -> Result<()> {
+    let _vscode = utils::require("code")?;
+
+    print_separator("Visual Studio Code");
+
+    run_type
+        .execute(shell())
+        .arg("-c")
+        .arg("for ext in $(code --list-extensions); do code --force --install-extension \"$ext\"; done")
```
Comment (r-darwish) [inline src/steps/generic.rs:96]: Thanks, but we need to avoid using a Unix shell because it won't work in Windows

**Review by r-darwish — CHANGES_REQUESTED** [review]

**Comment by r-darwish** [comment]
Will this download only new versions of packages or will it cause vscode to redownload all of your extensions every time Topgrade is invoked?

**Comment by SuperSandro2000** [comment]
> Will this download only new versions of packages or will it cause vscode to redownload all of your extensions every time Topgrade is invoked?

it will query the vscode marketplace for the latest version and if that is already installed nothing happens. This will take some time depended on the amount of extensions you have installed. For 43 extensions this took me 60 seconds.

File: src/steps/generic.rs (hunk 1 above)
Comment (AlexanderWillner) [reply src/steps/generic.rs:96]: What's the best practice here? Implementing the needed loop in rust I assume?

**Pull request merged**